control/control.go

```diff
@@ -337,6 +337,11 @@ func (c *Controller) Solve(ctx context.Context, req *controlapi.SolveRequest) (*
 		}
 	}
 
+	// if BUILDKIT_BUILDINFO build-arg is set, enforce it for the exporter
+	if v, ok := req.FrontendAttrs["build-arg:BUILDKIT_BUILDINFO"]; ok && v != "" {
+		req.ExporterAttrs["buildinfo"] = v
```

Comment:
does this mean it was possible to disable buildinfo before by setting `-o buildinfo=false`?

I think we should also check that `req.ExporterAttrs["buildinfo"]` was not already set.

Reply:
> does this mean it was possible to disable buildinfo before by setting `-o buildinfo=false`?

Yes https://github.com/moby/buildkit#imageregistry

> I think we should also check that `req.ExporterAttrs["buildinfo"]` was not already set.

Right `req.ExporterAttrs["buildinfo"]` should take precedence. Will make the change.